shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -673,12 +679,17 @@ private void initializePlatformViewIfNeeded(int viewId) {
     platformViews.put(viewId, view);
 
     FlutterMutatorView mutatorView =
-        new FlutterMutatorView(context, context.getResources().getDisplayMetrics().density);
+        new FlutterMutatorView(
+            context, context.getResources().getDisplayMetrics().density, androidTouchProcessor);
     mutatorViews.put(viewId, mutatorView);
     mutatorView.addView(platformView.getView());
     ((FlutterView) flutterView).addView(mutatorView);
   }
 
+  public void attachToFlutterRenderer(FlutterRenderer flutterRenderer) {
+    androidTouchProcessor = new AndroidTouchProcessor(flutterRenderer, /*trackMotionEvents=*/ true);
```

Comment:
wdyt about making `AndroidTouchProcessor` a singleton, and pass this `trackMotionEvents` to `onTouchEvent`?

Reply:
so for moving trackMotionEvents to AndroidTouchProcessor, the only reason i did not have the additional arg `onTouch`  was because, we also need it for `onGenericMotionEvent`, so Ideally, we would have a `MotionEventTrackingAndroidTouchProcessor` as a delegate for `AndroidTouchProcessor` and have it track all events that go through that, but I added an extra constructor param for now.